rust/navigator/src/navstate.rs

```diff
@@ -610,954 +602,1305 @@ impl State {
+                    let (content, ttype, author_info, network_info) = match t.action() {
+                        TransactionAction::Derivations {
+                            content,
+                            network_info,
+                            ..
+                        } => (
+                            content,
+                            TransactionType::ImportDerivations,
+                            None,
+                            Some(network_info),
+                        ),
+                        TransactionAction::Sign {
+                            content,
+                            author_info,
+                            network_info,
+                            ..
+                        } => (
+                            content,
+                            TransactionType::Sign,
+                            Some(author_info),
+                            Some(network_info),
+                        ),
+                        TransactionAction::Stub { s, .. } => (s, TransactionType::Stub, None, None),
+                        TransactionAction::Read { r } => (r, TransactionType::Read, None, None),
+                    };
+                    ScreenData::Transaction {
+                        f: MTransaction {
+                            content,
+                            ttype,
+                            author_info,
+                            network_info: network_info.map(|i| MSCNetworkInfo {
+                                network_title: i.title,
+                                network_logo: i.logo,
+                            }),
                         },
-                        transaction_parsing::Action::Stub(content, _, _) => format!("\"content\":{{{}}},\"type\":\"stub\"", content),
-                        transaction_parsing::Action::Read(content) => format!("\"content\":{{{}}},\"type\":\"read\"", content),
                     }
-                },
+                }
                 Screen::SeedSelector | Screen::SelectSeedForBackup => {
-                    match db_handling::interface_signer::print_all_seed_names_with_identicons(dbname, &self.seed_names) {
-                        Ok(a) => format!("\"seedNameCards\":{}", a),
-                        Err(e) => {
-                            new_navstate.alert = Alert::ErrorDisplay;
-                            errorline.push_str(&<Signer>::show(&e));
-                            "\"seedNameCards\":[]".to_string()
-                        },
-                    }
-                },
+                    let seed_name_cards =
+                        db_handling::interface_signer::get_all_seed_names_with_identicons(
+                            dbname,
+                            &self.seed_names,
+                        )
+                        .unwrap();
+                    let f = MSeeds { seed_name_cards };
+                    ScreenData::SeedSelector { f }
```

Comment:
Thus if screen is SelectSeedForBackup, screen data should be SelectSeedForBackup.

Reply:
This used to work when we had separate `Screen` variable; not anymore